packages/runtime/src/binding/ast.ts

```
@@ -798,7 +866,13 @@ export class PrimitiveLiteral<TValue extends StrictPrimitive = StrictPrimitive>
   public $kind: ExpressionKind.PrimitiveLiteral;
   public connect: IExpression['connect'];
   public assign: IExpression['assign'];
-  constructor(public readonly value: TValue) { }
+  public readonly value: TValue;
+
+  constructor(value: TValue) {
+    this.$kind = ExpressionKind.PrimitiveLiteral;
+    this.connect = PLATFORM.noop;
```

I think you're missing assign here?

---

Look all the way at the bottom of the file, I'm looping through all the AST classes and assigning `PLATFORM.noop` to `assign` and `connect` if they're falsey. You'll want to remove that bit of code as it's redundant when assigning things in the constructor. It should cause some tests to fail when you remove that and leave `assign` undefined.

I should note that `IExpression` is a pretty crappy interface. It's one of the first things that was migrated from vCurrent and it was done so in a very quick-and-dirty manner due to the sheer volume of code that had to be migrated. It's due for some refinements/additions. Something among the lines of `IAssignableExpression extends IExpression { assign }`, etc. Not sure yet.

In any case, `assign` shouldn't be optional. The AST should either do something, do nothing, or throw, but the function definitely needs to be present everywhere.